core/src/test/scala/org/apache/spark/util/collection/ExternalAppendOnlyMapSuite.scala

```diff
@@ -457,7 +458,7 @@ class ExternalAppendOnlyMapSuite extends SparkFunSuite
     // https://github.com/scala/scala/blob/2.13.x/test/junit/scala/tools/testing/AssertUtil.scala
     // (lines 69-89)
     // assert(map.currentMap == null)
-    eventually {
+    eventually(timeout(5 seconds), interval(200 milliseconds)) {
```

Comment:
I think best practice to encounter this is specifying config patience in sbt's test options.
having that said, I'v once had to 'grant' a longer timeout to a specific class so I've achieved this by overriding the `spanScalaeFactor` method

`override  def spanScaleFactor: Double = super.spanScaleFactor * 3`

please notice that there's another usage of `eventually` in line 519, this one 'manually' waits 500 millis before testing which might explain why you didn't see it failing in CI, looking at it now it seems like a bad practice since `eventually` is designed to control both the timeout and the intervals between trying. 

to summarize: best practice is to control this in sbt's test settings, if needed you can further control it in a specific class, in any case you have to make sure you handle all invocations of `eventually` (which is easier and less error prone by leveraging scalaTest's mechanisms).

Reply:
@dongjoon-hyun, thanks for cleaning up my mess! :sunglasses:

Reply:
Thank you for review, @eyalfa .

If you are looking at Spark code, you must notice that the most test cases are designed to have their suitable timeout and interval. We don't use a long global value for all test cases. That will hide potential big issues.

Also, I already knew that the other instance you mentioned, and 4 instance more on `ProcessingTimeExecutorSuite`. However, I didn't want to change it because I usually don't touch the test cases if it's not flaky.

Specifically, this instance is really serious in our Jenkins environment. The above 7 failures are only recent samples from 2 test series. We have additional 3 test series on branch `master`; maven-hadoop2.6, maven-hadoop2.7, sbt-scala-2.12. For `branch-2.4`, we have 4 more.

Reply:
This is a tiny issue annoying us. It's not a mess. :) Thank you again for your quick response.

Reply:
+1 for not overkilling it. If this is the only flaky test, let's fix this instance